### PR TITLE
NIFI-5656 Remove "Node Group" from the default FileAccessPolicyProvider propert…

### PR DESCRIPTION
**Pull request by pepov**
…er properties, extend the docs, add more logs to help debugging, add test for the empty case.

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [x] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pepov** [comment]
I didn't think about considering an empty group as not set, but now I can see that initial admin is treated the same, so will update the PR to work like that.

I used custom because I was thinking non of the available UGPs match that criteria. FileUserGroupProvider needs to be extended and wouldn't add that much benefit (nodes would still need to be added manually), however we may consider it if there are valid use cases.

However you are right with LdapUserGroupProvider and I missed that it loads first time synchronously and starts the background thread right after, my bad and thanks for pointing it out. I referred to configuration time because I wanted to emphasize that the initial user/group loading should be done in the `onConfigured` method, not later.  Of course that is how it's done by the LdapUserGroupProvider already, without that, the initial admin setup wouldn't work reliable.

**Comment by kevdoran** [comment]
Thanks @pepov! I'll re-review the PR when your changes are in.

**Comment by pepov** [comment]
@kevdoran changes are in

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-file-authorizer/src/main/java/org/apache/nifi/authorization/FileAccessPolicyProvider.java [hunk 1]
```diff
@@ -232,16 +232,21 @@ public void onConfigured(AuthorizerConfigurationContext configurationContext) th
...
-                }
 
-                if (nodeGroupIdentifier == null) {
-                    throw new AuthorizerCreationException(String.format(
+                    if (nodeGroupIdentifier == null) {
+                        throw new AuthorizerCreationException(String.format(
                             "Authorizations node group '%s' could not be found", nodeGroupName));
+                    }
+                } else {
+                    logger.warn("Empty node group name provided");
```
Comment (kevdoran) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-file-authorizer/src/main/java/org/apache/nifi/authorization/FileAccessPolicyProvider.java]: I don't think this merits a warning, given that it empty by default. Perhaps a debug message would be appropriate.

**Comment by kevdoran** [comment]
Thansk @pepov. I had one minor comment regarding logging, but aside from that these changes look good to me

**Comment by pepov** [comment]
Fixed that and also a checkstyle error that broke the build, hopefully will pass this time, thanks!

**Comment by kevdoran** [comment]
+1 thanks! Merged to master...
